e2e/support/helpers/e2e-embedding-iframe-sdk-helpers.ts

```diff
@@ -274,3 +276,36 @@ export const visitCustomHtmlPage = (
 
   return cy;
 };
+
+/**
+ * Mock the embed.js file to point it to the rspack dev server when not in CI.
+ *
+ * When running E2E tests locally we can usually benefit from hot reload.
+ * When we're testing something that uses the embed.js file though that file is always referenced
+ * from the uberjar so if we change it, we'd need to build the uberjar again.
+ *
+ * This function checks if the rspack dev server is available and if so, it mocks the embed.js file
+ * to point it to the rspack dev server.
+ */
+const mockEmbedJsToDevServer = () => {
+  if (Cypress.env("CI")) {
+    // we don't need this logic in CI, let's skip the check to avoid slowing down the tests
+    return;
+  }
+
+  // We use `cy.exec` with curl because both `fetch` and `cy.request` have downsides:
+  // - if we use `fetch`, it will show up as failed request in the browser polluting the logs
+  // - if we use `cy.request`, it will fail the test even if `failOnStatusCode` is false
+  cy.exec("curl --silent --head --fail http://localhost:8080/app/embed.js", {
+    failOnNonZeroExit: false,
+    log: false,
+  }).then((result) => {
+    const isHotReloadAvailable = result.code === 0;
```

Comment:
This is more like if the file is available at `http://localhost:8080/app/embed.js`, right? Since that `result` is curl's status code.